network/src/transport.rs

```diff
@@ -213,19 +213,40 @@ pub fn build_memory_noise_transport(
     application_protocols: SupportedProtocols,
 ) -> boxed::BoxedTransport<Connection<impl TSocket>, impl ::std::error::Error> {
     let memory_transport = memory::MemoryTransport::default();
-    let noise_config = Arc::new(NoiseConfig::new(identity_key));
+    let noise_config = Arc::new(NoiseWrapper::new(identity_key));
     let mut own_handshake = HandshakeMsg::new();
     own_handshake.add(SUPPORTED_MESSAGING_PROTOCOL, application_protocols);
 
     memory_transport
-        .and_then(move |socket, _addr, origin| async move {
-            let (remote_static_key, socket) =
-                noise_config.upgrade_connection(socket, origin).await?;
+        .and_then(move |socket, addr, origin| async move {
+            let remote_public_key = match addr.find_noise_proto() {
+                Some(public_key) => public_key,
+                None => {
+                    return Err(io::Error::new(
+                        io::ErrorKind::Other,
+                        "SHOULD NOT HAPPEN: remote public key not found",
+                    ))
+                }
+            };
+            let (remote_static_key, socket) = noise_config
+                .upgrade_connection(socket, origin, remote_public_key, Some(&trusted_peers))
+                .await?;
 
             // TODO(philiphayes): reenable after seed peers are always fully rendered
```

Comment:
How is this going to be the remote peers fully fleshed address? I think we still need to keep around a list of valid public keys and ensure that the authed pubkey is in the list.